src/index.js

```diff
@@ -113,7 +115,7 @@ export default function({ template, types, traverse }) {
               plugin = plugin.default
             }
 
-            return plugin({ template, types }).visitor
+            return plugin(api).visitor
```

Comment:
Re-implementing the plugin API is a bad idea, this is still not right. Plugins have 3 arguments. This should at least be

    export default function(api) {
    // instead as
    export default function(api, options, dirname) {

then

    return plugin(api, {}, dirname).visitor

and even then, since you're only taking `visitor` it's entirely possible that you're throwing away important flags from the plugin itself.

This might work for some plugins, but I'd expect it to fail more often as we land new features in Babel 7.